### ISSUE TITLE
🐛 [firebase_core_web,cloud_firestore_web] Build is broken after v1.0.1

### ISSUE THREAD

**Issue by MichaelDark** [FirebaseExtended/flutterfire#5436]
## Bug report

Cannot build app after upgrading to `cloud_firestore_web: 1.0.1` and `firebase_core_web: 1.0.1`, work only with overrides:

```
dependency_overrides:
  cloud_firestore_web: 1.0.0
  firebase_core_web: 1.0.1

dependencies:
  cloud_functions: ^1.0.1
  cloud_firestore: ^1.0.1
  firebase_core: ^1.0.1
```

Probably, broken after [this changes in PR#5335](https://github.com/FirebaseExtended/flutterfire/pull/5335) and [this release PR#5429](https://github.com/FirebaseExtended/flutterfire/pull/5429)

Output:
```
/opt/hostedtoolcache/flutter/2.0.2-stable/x64/.pub-cache/hosted/pub.dartlang.org/cloud_firestore_web-1.0.1/lib/src/interop/utils/utils.dart:15:41: Error: The argument type 'Object? Function(Object)' can't be assigned to the parameter type 'Object? Function(Object?)?' because 'Object?' is nullable and 'Object' isn't.

 - 'Object' is from 'dart:core'.

  return core_interop.dartify(jsObject, (Object object) {

                                        ^

/opt/hostedtoolcache/flutter/2.0.2-stable/x64/.pub-cache/hosted/pub.dartlang.org/cloud_firestore_web-1.0.1/lib/src/interop/utils/utils.dart:39:41: Error: The argument type 'dynamic Function(Object)' can't be assigned to the parameter type 'Object? Function(Object?)?' because 'Object?' is nullable and 'Object' isn't.
 - 'Object' is from 'dart:core'.


  return core_interop.jsify(dartObject, (Object object) {

                                        ^
```

### Steps to reproduce

1. Add `cloud_firestore: ^1.0.1`- to application;
2. Build application;
3. Receive build error.

### Expected behavior

Build finished without error.
---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.0.1, on macOS 11.2.3 20D91 darwin-arm, locale en-UA)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] IntelliJ IDEA Community Edition (version 2020.3.2)
[✓] Connected device (1 available)

• No issues found!
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 2.12.0
Flutter SDK 2.0.1

dependencies:
- cloud_firestore 1.0.1 [cloud_firestore_platform_interface cloud_firestore_web firebase_core firebase_core_platform_interface flutter meta]
- cloud_functions 1.0.1 [cloud_functions_platform_interface cloud_functions_web firebase_core firebase_core_platform_interface flutter]
- firebase_core 1.0.2 [firebase_core_platform_interface firebase_core_web flutter meta]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_localizations 0.0.0 [flutter intl characters clock collection meta path typed_data vector_math]

dev dependencies:
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]
- test 1.16.5 [analyzer async boolean_selector coverage http_multi_server io js node_preamble package_config path pedantic pool shelf shelf_packages_handler shelf_static shelf_web_socket source_span stack_trace stream_channel typed_data web_socket_channel webkit_inspection_protocol yaml test_api test_core]

transitive dependencies:
- _fe_analyzer_shared 18.0.0 [meta]
- analyzer 1.2.0 [_fe_analyzer_shared cli_util collection convert crypto glob meta package_config path pub_semver source_span watcher yaml]
- args 2.0.0
- async 2.5.0 [collection]
- boolean_selector 2.1.0 [source_span string_scanner]
- characters 1.1.0
- charcode 1.2.0
- cli_util 0.3.0 [meta path]
- clock 1.1.0
- cloud_firestore_platform_interface 4.0.0 [collection firebase_core flutter meta plugin_platform_interface]
- cloud_functions_platform_interface 5.0.0 [firebase_core flutter meta plugin_platform_interface]
- cloud_functions_web 4.0.2 [cloud_functions_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- collection 1.15.0
- convert 3.0.0 [typed_data]
- coverage 1.0.1 [args logging package_config path source_maps stack_trace vm_service]
- crypto 3.0.0 [collection typed_data]
- fake_async 1.2.0 [clock collection]
- ffi 1.0.0
- file 6.1.0 [meta path]
- firebase_core_platform_interface 4.0.0 [flutter meta plugin_platform_interface]
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- glob 2.0.0 [async collection file path pedantic string_scanner]
- http 0.13.0 [http_parser meta path pedantic]
- http_multi_server 3.0.0 [async]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- io 1.0.0 [meta path string_scanner]
- js 0.6.3
- logging 1.0.0
- matcher 0.12.10 [stack_trace]
- meta 1.3.0
- mime 1.0.0
- node_preamble 1.4.13
- package_config 2.0.0 [path]
- package_info 2.0.0 [flutter]
- path 1.8.0
- path_provider 2.0.1 [flutter path_provider_platform_interface path_provider_macos path_provider_linux path_provider_windows]
- path_provider_linux 2.0.0 [path xdg_directories path_provider_platform_interface flutter]
- path_provider_macos 2.0.0 [flutter]
- path_provider_platform_interface 2.0.1 [flutter meta platform plugin_platform_interface]
- path_provider_windows 2.0.0 [path_provider_platform_interface meta path flutter ffi win32]
- platform 3.0.0
- plugin_platform_interface 2.0.0 [meta]
- pool 1.5.0 [async stack_trace]
- process 4.1.0 [file path platform]
- pub_semver 2.0.0 [collection]
- shelf 1.1.0 [async collection http_parser path stack_trace stream_channel]
- shelf_packages_handler 3.0.0 [path shelf shelf_static]
- shelf_static 1.0.0 [convert http_parser mime path shelf]
- shelf_web_socket 1.0.1 [shelf stream_channel web_socket_channel]
- sky_engine 0.0.99
- source_map_stack_trace 2.1.0 [path stack_trace source_maps]
- source_maps 0.10.10 [source_span]
- source_span 1.8.0 [charcode collection path term_glyph]
- stack_trace 1.10.0 [path]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- test_api 0.2.19 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- test_core 0.3.15 [analyzer async args boolean_selector collection coverage glob io meta package_config path pedantic pool source_map_stack_trace source_maps source_span stack_trace stream_channel vm_service yaml matcher test_api]
- typed_data 1.3.0 [collection]
- url_launcher_linux 2.0.0 [flutter]
- url_launcher_macos 2.0.0 [flutter]
- url_launcher_platform_interface 2.0.2 [flutter plugin_platform_interface]
- url_launcher_web 2.0.0 [url_launcher_platform_interface meta flutter flutter_web_plugins]
- url_launcher_windows 2.0.0 [flutter]
- uuid 3.0.1 [crypto]
- vector_math 2.1.0
- vm_service 6.1.0+1
- watcher 1.0.0 [async path pedantic]
- web_socket_channel 2.0.0 [async crypto stream_channel]
- webkit_inspection_protocol 1.0.0 [logging]
- win32 2.0.4 [ffi]
- xdg_directories 0.2.0 [meta path process]
- yaml 3.1.0 [collection source_span string_scanner]
```

</details>

---

**Comment by MichaelDark** [comment]
@Salakar Why CI on release commit failed, but packages had been released?

**Comment by markusaksli-nc** [comment]
Hi @MichaelDark 
These changes were introduced in `cloud_firestore 1.0.2` (`cloud_firestore_web 1.0.2`). That being said I'm not seeing this on either `1.0.1` or `1.0.2` on stable or master.
Is something missing from the repro steps? Not seeing any red flags on https://pub.green/packages/cloud_firestore_web
Thank you

**Comment by rrousselGit** [comment]
> Cannot build app after upgrading to `cloud_firestore_web: 1.0.1` and `firebase_core_web: 1.0.1`,

But the versions that were published aren't 1.0.1 but 1.0.2

I too am unable to reproduce this issue. Try upgrading your packages to the latest version available and make sure to run `dart pub upgrade`.

**Comment by MichaelDark** [comment]
Thank you, after running `flutter pub upgrade`, issue resolved. It was because somehow `pubspec.lock` file end up with incompatible versions for firebase ecosystem.
